add_nyt_bestseller_tag.py

```diff
@@ -20,87 +20,107 @@
+                for bstslr_record_isbn in bestseller_group_record['isbns']:
+                    try:
+                        bstslr_edition = self.ol.Edition. \
+                            get(isbn=bstslr_record_isbn)
+                        if bstslr_edition:
                             self.logger.info('The edition {} exists in OL'
-                                             .format (bstslr_record_isbn))
-                            bstslr_edition = self.ol.Edition.get(
-                                isbn=bstslr_record_isbn)
-                            bstslr_work = bstslr_edition.work()
-                            if self.need_to_add_nyt_bestseller_tag(bstslr_work):
+                                             .format(bstslr_record_isbn))
+                            if self.need_to_add_nyt_bestseller_tag(
+                                bstslr_edition.work):
                                 self.logger.info(
-                                    'The NYT tag to be added for edition {}'
-                                        .format(bstslr_record_isbn))
-                                self.add_tag(bstslr_work, new_tag)
-                                bstslr_work.save(comment)
+                                    'The NYT tag to be added for the work {}'
+                                    ' of the edition {}'
+                                        .format(bstslr_edition.work.olid,
+                                                bstslr_record_isbn))
+                                self.add_tag(bstslr_edition.work, new_tag)
+                                bstslr_edition.work.save(comment)
                                 bstslr_edition.save(comment)
+                                job_results['tags_added'] = \
+                                    job_results['tags_added'] + 1
+                            else:
+                                self.logger.info(
+                                    'The NYT tag already exists for the work {}'
+                                    ' of the edition {}, skipping {}'
+                                        .format(bstslr_edition.work.olid,
+                                                bstslr_record_isbn,
+                                                repr(bstslr_edition.work)))
+                                job_results['tags_already_exist'] = \
+                                    job_results['tags_already_exist'] + 1
                         else:
                             self.logger.info(
                                 'The edition {} doesnt exist in OL, importing'
                                     .format(bstslr_record_isbn))
                             self.request_book_import_by_isbn(bstslr_record_isbn)
+                            job_results['books_imported'] = \
+                                job_results['books_imported'] + 1
+                    except:
+                        self.logger.exception('Failed to process ISBN {}'
```

Comment:
Could we keep track of a list of failed isbns to share at the end?

Reply:
Will add the failed ISBNs counter, no problem!